### PR TITLE
feat(content): Hai Merchant Expansion

### PR DESCRIPTION
**Pull request by MasterOfGrey**
**Content (Worldbuilding)**

## Summary
This PR introduces a number of new Hai governments that allow for greater granularity of behaviour with respect to the Hai and the Unfettered, lays groundwork for future plot, and (should hopefully) reduce the all-out war status of Wah Ki. Separating out security groups from merchants - though note a couple of transports and freighters still exist with military escort in the government fleets.

Further it generally increases the number of ships flying around in Hai space to ~double what is currently true. There's a lot of Hai crammed in a small space, it should be bustling, now it should feel like it.

Thirdly it introduces a Hai (Human) merchant fleets, which represent x generation humans who have been born and raised in Hai space, and whose fleets are consequently registered in Hai space not human space, these fleets are a mix of Hai ships and legacy human ships with Hai outfits.

Lastly it moves all the fleets into their own file because `hai.txt` was getting unwieldy. 

## Save File
Any save file can be used to check it out.

### CODE REVIEW

**Comment by Saugia** [comment]
Does this mean there will be no Hai war system anymore? Or at least any systems near the level of Wah Ki? That is almost a second Mesuket in the game with how much battle happens. On the same line, though, if that was the case, it would lower the consistency of the "go wait and capture disabled unfettered beetles and win the game" strategy from early on, which I think is definitely a good thing.

**Comment by MasterOfGrey** [comment]
> Does this mean there will be no Hai war system anymore? Or at least any systems near the level of Wah Ki? That is almost a second Mesuket in the game with how much battle happens. On the same line, though, if that was the case, it would lower the consistency of the "go wait and capture disabled unfettered beetles and win the game" strategy from early on, which I think is definitely a good thing.

The fighting in Wah Ki is definitely not gone yet. Unlike before though, a large portion of the Hai ships will ignore it and carry on with their business though, because the (Sympathizer) merchants aren't hostile to the Unfettered.

**Comment by Terin** [comment]
At least Hai Merchant and Sympathizers should be added to the Wormhole Alpha government's "hostile" list, right?

**Comment by MasterOfGrey** [comment]
> At least Hai Merchant and Sympathizers should be added to the Wormhole Alpha government's "hostile" list, right?

They're not set to spawn in the system with the wormhole, so it's not strictly necessary.

**Comment by Terin** [comment]
With as tightly packed as Hai space is, though, they can easily jump in if you're leaving through the Hai side. Probably worth covering that, at least.

**Comment by McloughlinGuy** [comment]
You should make some hails for them

**Comment by MasterOfGrey** [comment]
> You should make some hails for them.

I assume you mean for the Hai merchants that are actually nth generation humans?
If so, yeah probably. I think they can get by just fine for now though, I'd prefer to tackle an expansion of their hails in a subsequent PR rather than here.

**Comment by McloughlinGuy** [comment]
> > You should make some hails for them.
> 
> I assume you mean for the Hai merchants that are actually nth generation humans? If so, yeah probably. I think they can get by just fine for now though, I'd prefer to tackle an expansion of their hails in a subsequent PR rather than here.

I actually meant both, but making hails for human merchants would be a much higher priority than making hails for hai merchants.

**Comment by MasterOfGrey** [comment]
> I actually meant both, but making hails for human hai merchants would be a much higher priority than making hails for hai merchants.

But... they both just use existing hails which already exist for Hai and Merchants.
It's not like there's a vacuum to fill here. There's a space where an improvement can be made, but there's valid content already there.

Hence why I would do it in a subsequent PR, not this one.

**Review by Hecter94 — APPROVED** [review]
I gave this PR a pretty thorough run-through, focusing on Hai space. For testing purposes I immediately moved to Hai space after starting a new game, and stayed in Hai space for the remainder of the game unless directed outside of Hai space by a mission. 
During this time, I used every Hai ship commercially available, as well as a Shield Beetle which is where I ended the run. 

A couple of things I noticed which I would like to note:
1) Hai space is most definitely far more crowded and bustling than it was before. There are many, large fleets moving between systems and you see a lot more ships in general

2) In reference to the point above, trying to get through Hai space with any illegal goods is now nearly impossible. You are scanned and fined constantly. I believe this is thematically appropriate given Hai society and I don't think it has much of an effect on gameplay either, so if this was an intentional change, I support it. 

3) The Hai fleets near the wormhole never seem willing to refuel my ships, in vanilla I can get a refuel after coming out of the wormhole but before entering Hai space, with this PR installed, I ended up needing to reload a save, install an additional fuel pod and try again. This doesn't seem appropriate for the Hai, and it means that the "wonder" of discovering this new region of space with an alien species can get lost if the player immediately needs to reload a save because they got stranded one jump away. 

4) I found that capturing a shield beetle was made overall easier with these changes. There was less fighting in Wah Ki so I did have to wait longer before getting a disabled one, but it was also far less risky due to the smaller number of ships in the system. I believe this would result in making it easier, but more time consuming to capture shield beetles which does reduce the value of "farming" them for credits. Doubly so as nerve gas is now a huge liability in Hai space.

Overall, I am going to say that I approve of this PR, as I believe it does many things well. There are a couple of things mentioned above that I would personally like to see looked into, but overall I approve of this PR being merged.

**Comment by MasterOfGrey** [comment]
This review is greatly appreciated.

> 1. Hai space is most definitely far more crowded and bustling than it was before. There are many, large fleets moving between systems and you see a lot more ships in general

Good that was intentional.

> 2. In reference to the point above, trying to get through Hai space with any illegal goods is now nearly impossible. You are scanned and fined constantly. I believe this is thematically appropriate given Hai society and I don't think it has much of an effect on gameplay either, so if this was an intentional change, I support it.

That is an unexpected but positive side effect. I imagine that this will also get revisited when the effort to make fining illegals an atrocities government specific too so I suspect this is not a problem.

> 3. The Hai fleets near the wormhole never seem willing to refuel my ships, in vanilla I can get a refuel after coming out of the wormhole but before entering Hai space, with this PR installed, I ended up needing to reload a save, install an additional fuel pod and try again. This doesn't seem appropriate for the Hai, and it means that the "wonder" of discovering this new region of space with an alien species can get lost if the player immediately needs to reload a save because they got stranded one jump away.

Ah now this I'll look into.

> 4. I found that capturing a shield beetle was made overall easier with these changes. There was less fighting in Wah Ki so I did have to wait longer before getting a disabled one, but it was also far less risky due to the smaller number of ships in the system. I believe this would result in making it easier, but more time consuming to capture shield beetles which does reduce the value of "farming" them for credits. Doubly so as nerve gas is now a huge liability in Hai space.

A pleasant side effect.

I'm not sure why Hai fleets near the wormhole wouldn't refuel you. If anyone knows why please suggest a fix for it, otherwise I'll try and work it out at some point soonish.

**Comment by Hecter94** [comment]
> I'm not sure why Hai fleets near the wormhole wouldn't refuel you. If anyone knows why please suggest a fix for it, otherwise I'll try and work it out at some point soonish.

I looked into this a bit more, to try to troubleshoot the problem myself and couldn't replicate it. 
I'm not sure what happened as when I was trying the PR previously, I got stuck at the wormhole on two separate occasions and none of the Hai ships passing through were willing to refuel me even after spending a while waiting for various different fleets to pass through.

However I just loaded up this PR again on the same save with the intent of looking into this issue, and had no problems getting a Hai ship to refuel me in Waypoint multiple times.

I'm thinking it may have been a bug unrelated to this PR, though I can't say for sure. At the very least, it does not appear to be a global issue.

**Comment by MasterOfGrey** [comment]
That's good to know

**Comment by MasterOfGrey** [comment]
@W1zrad I think this is up to you now, at your convenience. Responses may be delayed on my part - you have permission to make minor adjustments directly without my feedback.

I would suspect that the biggest chance of needing further edits will be if someone brings up a particular system as feeling too busy or too empty. <- I used a strict ratio for all my fleet changes for every system except those right next to Wah Ki. So there's no guarantee that the old traffic ratios feel perfectly alright with the increased ship numbers overall. That's the main thing I recommend eyeballing in your review.

**Comment by roadrunner56** [comment]
Agreed. I never got the impression that the average Hai was stuck to one planet their whole life. The Hai are stagnant technologically, but most Hai are shown to be better off than the average human. Considering this, it would make sense that there's still a lot of interstellar commerce and travel.

**Comment by W1zrad** [comment]
Been thinking about this the past few days and my opinion has swung, especially considering there's the mention of only a small proportion of humans leaving planets and flying ships anyway. So consider that concern resolved!

**Comment by W1zrad** [comment]
Sorry, life's quite busy right now and will be until mid-May. I still entirely intend to swing around to this as soon as I'm able, but I want to leave this open to other folks in case someone else wants to come through and merge it.

File: data/hai/hai fleets.txt [hunk 1]
```diff
@@ -0,0 +1,572 @@
...
+	variant 3
+		"Grasshopper (Surveillance)" 5
+
+
+fleet "Small Hai Merchant"
+	government "Hai"
+	names "hai"
+	cargo 3
+	personality
+		confusion 15
```
Comment (W1zrad) [inline data/hai/hai fleets.txt:143]:
```suggestion
		confusion 35
```

File: data/hai/hai fleets.txt [hunk 2]
```diff
@@ -0,0 +1,572 @@
...
+	variant 1
+		"Grasshopper"
+		"Centipede" 2
+
+fleet "Large Hai Merchant"
+	government "Hai"
+	names "hai"
+	cargo 3
+	personality
+		confusion 15
```
Comment (W1zrad) [inline data/hai/hai fleets.txt:205]:
```suggestion
	cargo 4
	personality
		confusion 35
```

File: data/hai/hai fleets.txt [hunk 3]
```diff
@@ -0,0 +1,572 @@
...
+		"Water Bug"
+		"Grasshopper (Tracker)"
+
+
+fleet "Small Hai Merchant (Sympathizers)"
+	government "Hai Merchant (Sympathizers)"
+	names "hai"
+	cargo 3
+	personality
+		confusion 15
```
Comment (W1zrad) [inline data/hai/hai fleets.txt:245]:
```suggestion
		confusion 35
```

File: data/hai/hai fleets.txt [hunk 4]
```diff
@@ -0,0 +1,572 @@
...
+	variant 1
+		"Grasshopper"
+		"Centipede" 2
+
+fleet "Large Hai Merchant (Sympathizers)"
+	government "Hai Merchant (Sympathizers)"
+	names "hai"
+	cargo 3
+	personality
+		confusion 15
```
Comment (W1zrad) [inline data/hai/hai fleets.txt:307]:
```suggestion
	cargo 4
	personality
		confusion 35
```

File: data/hai/hai fleets.txt [hunk 5]
```diff
@@ -0,0 +1,572 @@
...
+		"Water Bug"
+		"Grasshopper (Tracker)"
+
+
+fleet "Small Hai Merchant (Human)"
+	government "Hai Merchant (Human)"
+	names "hai"
+	cargo 3
+	personality
+		confusion 15
```
Comment (W1zrad) [inline data/hai/hai fleets.txt:347]:
```suggestion
		confusion 35
```

File: data/hai/hai fleets.txt [hunk 6]
```diff
@@ -0,0 +1,572 @@
...
+	variant 1
+		"Star Barge"
+		"Aphid"
+
+fleet "Large Hai Merchant (Human)"
+	government "Hai Merchant (Human)"
+	names "hai"
+	cargo 3
+	personality
+		confusion 15
```
Comment (W1zrad) [inline data/hai/hai fleets.txt:384]:
```suggestion
	cargo 4
	personality
		confusion 35
```

**Review by W1zrad — CHANGES_REQUESTED** [review]
We back.
Everything here looks solid, only got two thoughts:
1. `confusion` numbers for all these new merchant fleets (15) are quite low. Per the wiki:

> The confusion value is the maximum error in pixels, and defaults to 10. Generally I use a value of 10 for military ships, 20 for pirates, 30 for skilled merchants and 40 for ordinary merchants.

So we're shooting for somewhere between 30-40. Maybe 35 since they're near pirates but not 
often interacting with them (i.e. they're on the other side of the wormhole)?

2. `cargo` for the new fleets seems to be 3 across the board. Let's get some variety in there, no? For human fleets, cargo stats for merchants are 3 for small, 4 for large, so we'll use that.

These changes are implemented in the suggestions below.

**Review by MasterOfGrey — COMMENTED** [review]
> We back. Everything here looks solid, only got two thoughts:

And you got me at a good moment!

> 1. `confusion` numbers for all these new merchant fleets (15) are quite low. Per the wiki:
> 
> > The confusion value is the maximum error in pixels, and defaults to 10. Generally I use a value of 10 for military ships, 20 for pirates, 30 for skilled merchants and 40 for ordinary merchants.

Ok this has brought to my attention that the game does not reflect the wiki. A vast number of fleets have no confusion values attached when clearly, if that's the design philosophy, they should all have some.

> So we're shooting for somewhere between 30-40. Maybe 35 since they're near pirates but not often interacting with them (i.e. they're on the other side of the wormhole)?

So, every Hai pilot alive has either lived through a war, or likely been trained by someone who did. To my understanding the last generation from the Korath wars are still around, albeit now almost entirely retired. They and their direct trainees would have dominated the pilot training for every Hai alive. 

On top of that, the Unfettered conflict obviously exists, which would place passive pressure on skill levels.

Add to that that from Nanachi we know Hai ships are expensive, and that most people who own them come from legacies of ship-owners who would all have that recent generational history of war - and also that Hai live to have a solid couple hundred years of adulthood.

So I would expect their merchants to be considerably more disciplined than human merchants.

^Given all that I have added suggestions to change them all to 25 (split the difference between pirate and advanced merchant), and also added `confusion 5` to all Hai and Unfettered military fleets.

> 2. `cargo` for the new fleets seems to be 3 across the board. Let's get some variety in there, no? For human fleets, cargo stats for merchants are 3 for small, 4 for large, so we'll use that.

Makes sense.

I'm not gonna fight about confusion though, if you have a mind to enforce the wiki philosophy more aggressively and use 35 go ahead. You have my permission to batch up and commit whichever set of suggestions suits you. :)

**Comment by W1zrad** [comment]
Fair points!

> Ok this has brought to my attention that the game does not reflect the wiki. A vast number of fleets have no confusion values attached when clearly, if that's the design philosophy, they should all have some.

Whoops, probably should have also quoted from the wiki that the default value for confusion is 10. With that in mind, I'm cool with the merchant change to 25, but I'll leave the military ships where they are (mostly cause the automata, literal robots, are set at a default of 10 too). So maybe is confusion something that I/someone/we could look at and try to make uniform, but for now we'll leave it at that.

**Review by W1zrad — APPROVED** [review]

**Pull request merged**